### DISCUSSION TITLE
Allow qualified names to be used in property initializers

### DISCUSSION THREAD

**Discussion by michaelcheers** [dotnet/csharplang#5068] [Language Ideas]
Right Now:

```csharp
new HTMLDivElement { Style = { FontColor = "red" } }
```

New Syntax:

```csharp
new HTMLDivElement { Style.FontColor = "red" }
```

**Comment by HaloFour** [comment]
Similar is already being implemented for pattern matching and I believe the intent was to bring that over to "withers", so it does make sense to also extend that to object initializers, IMO.

See: https://github.com/dotnet/csharplang/issues/4394

**Reply by jnm2** [reply]
For `with` expressions: https://github.com/dotnet/csharplang/discussions/77
